### ISSUE TITLE
Heard + Genesis extremely slow

### ISSUE THREAD

**Issue by mikemastercorp** [thedevdojo/genesis#47]
Hello Team.

I was super glad to find another project of the dojo to help me with a quick setup of TALL+F+V and I've quickly got the repo, setup the project in my Laravel Herd Premium and set it all up. As promised it started working and I was glad that the setup went painlessly. 

However when loading any of the views, I must say it takes around 20-30 seconds before the page is shown and the tests were ran on a 12th gen Intel i7 CPU + 64GB Ram and Nvidia A3000 GPU where any other projects I ran with the same Herd usually took less than 1-2 sec to load.

Tried to check for some logs to figure out where this lag can come up but so far no clue. Any ideas or suggestions what could be going wrong as this PC has no other apps running besides my PHPStorm and is destined solely for my work...

Thanks anyway for the great package, I will still use it once I figure out what is happening...

**Comment by kriiv** [comment]
Have you checked debug bar? There's likely something else going on.

https://github.com/barryvdh/laravel-debugbar

**Comment by ikx94** [comment]
I have the exact same issue. Installed Laravel Debug Bar and can't figure it out.

At the moment it's not usable. I'm on Windows 11 and all other Laravel apps run fine.

**Comment by tnylea** [comment]
We've experienced this issue with a few other users. They were able to resolve it by running:

```
php artisan optimize
```

Please, run that and let us know. Closing for now, but feel free to open if it's still not working ;) Thanks!

**Comment by bobbyiliev** [comment]
In case that anyone hits this in the future, in addition to the `php artisan optimize` command, also try `php artisan filament:optimize`.

And as per the discussion here: https://github.com/blade-ui-kit/blade-icons/issues/71#issuecomment-1810493388, add the following middleware to your Laravel app:

```php
protected $middleware = [
    ...
    \Filament\Http\Middleware\DisableBladeIconComponents::class
];
```

**Comment by chriscupas** [comment]
> We've experienced this issue with a few other users. They were able to resolve it by running:
> 
> ```
> php artisan optimize
> ```
> 
> Please, run that and let us know. Closing for now, but feel free to open if it's still not working ;) Thanks!

I also experienced this slow performance issue when running Genesis in a container environment (Docker) and Kubernetes (K8s).

The issue seems to be related to how Volt handles view caching in these environments. I've submitted a fix to livewire/volt (PR #150) that addresses a TypeError when `Blade::getPath()` returns null during view caching, which can occur in containerized environments.

This fix is included in livewire/volt v1.10.2. I've also submitted a PR to Genesis to update the Volt dependency from ^1.7 to ^1.10.2 which may help resolve this issue.

**Related:**
- livewire/volt#150 (merged)
- thedevdojo/genesis#59 - updates Volt to ^1.10.2
